smartsim/_core/_install/buildenv.py

```diff
@@ -58,7 +69,11 @@ class SetupError(Exception):
     See setup.py for more
     """
 
-    pass
+
+class VersionConflictError(SetupError):
+    """An error for when version numbers of some library/package/program/etc
+    do not match and build may not be able to continue
+    """
```

Comment:
Is there any potential benefit to stashing name, current_version, target_version in this class? The message could be automatically generated from those three

Reply:
Brilliant! That also sounds like useful info should one want to catch and handle that err